mytoyota/api.py

```diff
@@ -1,140 +1,183 @@
+from mytoyota.models.endpoints.notifications import NotificationResponseModel
+from mytoyota.models.endpoints.status import RemoteStatusResponseModel
+from mytoyota.models.endpoints.telemetry import TelemetryResponseModel
+from mytoyota.models.endpoints.trips import TripsResponseModel
+from mytoyota.models.endpoints.vehicle_guid import VehiclesResponseModel
+from mytoyota.models.endpoints.vehicle_health import VehicleHealthResponseModel
 
-from typing import Any
 
-from .const import BASE_URL, BASE_URL_CARS
-from .controller import Controller
+class Api:
+    """API Class. Allows access to available endpoints to retrieve the raw data."""
 
+    def __init__(self, controller: Controller) -> None:  # noqa: D417
+        """Initialise the API.
 
-class Api:
-    """Controller class."""
+        Initialise the API and set the Controller
 
-    def __init__(self, controller: Controller) -> None:
-        """Toyota Controller"""
+        Parameters
+        ----------
+            controller: Controller: A controller class to managing communication
+        """
         self.controller = controller
 
-    @property
-    def uuid(self) -> str | None:
-        """Returns uuid from controller"""
-        return self.controller.uuid
+    async def _request_and_parse(self, model, method: str, endpoint: str, **kwargs):
+        """Parse requests and responses."""
+        response = await self.controller.request_json(method=method, endpoint=endpoint, **kwargs)
+        return model(**response)
 
-    async def set_vehicle_alias_endpoint(
-        self, new_alias: str, vehicle_id: int
-    ) -> dict[str, Any] | None:
-        """Set vehicle alias."""
-        return await self.controller.request(
+    async def set_vehicle_alias_endpoint(self, alias: str, guid: str, vin: str):
+        """Set the alias for a vehicle."""
+        return await self.controller.request_raw(
             method="PUT",
-            base_url=BASE_URL_CARS,
-            endpoint=f"/api/users/{self.uuid}/vehicles/{vehicle_id}",
-            body={"id": vehicle_id, "alias": new_alias},
+            endpoint=VEHICLE_ASSOCIATION_ENDPOINT,
+            vin=vin,
+            headers={
+                "datetime": str(int(datetime.now(timezone.utc).timestamp() * 1000)),
+                "x-correlationid": str(uuid4()),
+                "Content-Type": "application/json",
+            },
+            body={"guid": guid, "vin": vin, "nickName": alias},
         )
 
-    async def get_vehicles_endpoint(self) -> list[dict[str, Any] | None] | None:
-        """Retrieves list of cars you have registered with MyT"""
-        return await self.controller.request(
-            method="GET",
-            base_url=BASE_URL_CARS,
-            endpoint=f"/vehicle/user/{self.uuid}/vehicles?services=uio&legacy=true",
-        )
+    #    TODO: Remove for now as it seems to have no effect. The App is sending it!
+    #    async def post_wake_endpoint(self) -> None:

```

Comment:
What does this do?

Reply:
Nobody knows 😃

Reply:
I was hopeful it "woke up" the car and got the latest information, but nothing seems to update. It can be removed for now.